clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -58,6 +61,8 @@ public class AbstractConfig {
 
     private final ConfigDef definition;
 
+    public static final String AUTOMATIC_CONFIG_PROVIDERS_PROPERTY = "org.apache.kafka.automatic.config.providers";
```

Comment:
I'm not sure on the naming. What about `org.apache.kafka.allowed.config.providers`?

Reply:
I wanted to distinguish the fact that this only affects the "automatic resolution", not every use of the config providers, which is how I would interpret your suggested name. The "automatic" comes from the language of the original KIP, though i'm not sure how many people call this "automatic resolution"

Reply:
Looks like a public API change to me. Don't we need a KIP?